controller/array_action/ds8k_rest_client.py

```diff
@@ -0,0 +1,245 @@
+from pyds8k.client.ds8k.v1.client import Client
+from pyds8k.exceptions import NotFound
+from controller.common.csi_logger import get_stdout_logger
+
+logger = get_stdout_logger()
+
+
+def normalize_response(func):
+    def inner(self, *args, **kwargs):
+        res_obj = func(self, *args, **kwargs)
+        if not isinstance(res_obj, list):
+            res_obj = [res_obj, ]
+        return list(map(_normalize_item, res_obj))
+    return inner
+
+
+def _normalize_item(res):
+    try:
+        return res.representation
+    except AttributeError:
+        return res
+
+
+def _int_lunid_to_hex(lunid):
+    return '{0:0{1}x}'.format(int(lunid), 2)
+
+
+def int_to_scsilun(lun):
+    # There are two style lun number, one's decimal value is <256 and the other
+    # is full as 16 hex digit. According to T10 SAN, when decimal value is more
+    # then 256 and it is converted to the full 16 hex digit, it should be
+    # swapped and converted into hex.
+    # For example, zlinux lun number stored in SC DB is 1075331113. It should
+    # be hex to '40184029' and swapped to '40294018'. When the lun number is 12
+    # and it is <256, it should be hex to '0c' directly.
+    pretreated_lun = int(lun)
+    if pretreated_lun < 256:
+        return _int_lunid_to_hex(pretreated_lun)
+    else:
+        return '{0:x}'.format(
+            (pretreated_lun >> 16 & 0xFFFF) | (pretreated_lun & 0xFFFF) << 16
+        )
+
+
+def scsilun_to_int(lun):
```

Comment:
why did you resolve this?

Reply:
I will keep this function.

Reply:
where will it be used? I cannot tell since the code is not finished

Reply:
the code is not finished? Why did you say that?

Reply:
Because I assume you want to keep this function for code that does not exist yet (i.e. the current code does not call this function, or in other words - not finished)

Reply:
ok, understand. But it is important for z support, so I will keep it.

Reply:
But is it necessary for pvc creation in Z?

Reply:
It will be used later, like mount.

Reply:
OK.